pkg/controller/seed-controller-manager/cluster-template-controller/controller.go

```diff
@@ -188,17 +170,57 @@ func (r *reconciler) createClusters(ctx context.Context, instance *kubermaticv1.
 
 				return fmt.Errorf("failed to create desired number of clusters. Created %d of %d", created, totalReplicas)
 			}
```

Comment:
Don't we want update the number of replicas when we succeed creating a replica? 
Shouldn't this be outside of err != nil block? 

```
				created := int64(i + 1)
				totalReplicas := instance.Spec.Replicas

				if err := r.patchInstance(ctx, instance, func(i *kubermaticv1.ClusterTemplateInstance) {
					i.Spec.Replicas = totalReplicas - created
				}); err != nil {
					return err
				}
```

Reply:
What's the point in patching the ClusterTemplateInstance n times in a loop? Either the loop succeeds and we delete the ClusterTemplateInstance or it fails and we update the spec.

Reply:
1. Say instance.Spec.Replicas = 3 
2. r.createCluster fails on the first call. 

From the current code looks like 

instance.Spec.Replicas will be  2. It should be 3, right?

Reply:
No, it should be 2. Because on the next reconciliation, we only want to create the remaining 2 clusters. Otherwise we end up in an infinite loop.

Reply:
No, it should be 2.

@zreigz